### PR TITLE
Update to new user API.

### PR DESCRIPTION
**Pull request by mdwelsh**
This PR updates the `fixie` package to use the new `/api/v1/user/me` API for getting and updating details on the current user. This is needed so we can support this in the web UI.

### CODE REVIEW

**Comment by mdwelsh** [comment]
@petersalas This is now ready for a review. 

I am making some important structural changes here: The build goes in 'dist', and getting rid of 'web.ts' which is no longer needed as far as I can tell. Can you please look?

File: packages/fixie/.gitignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-*.js
-*.d.ts
```
Comment (petersalas) [inline packages/fixie/.gitignore:2]: why did you remove this?

File: packages/fixie/tsconfig.json [hunk 2]
```diff
@@ -9,10 +9,11 @@
...
     "jsx": "react",
     "esModuleInterop": true,
-    "outDir": ".",
+    "outDir": "./dist",
+    "rootDir": "./src",
     "declaration": true,
     "lib": ["dom", "dom.iterable", "ES2022"]
   },
-  "include": ["src/*", "jest.config.ts", ".eslintrc.cjs", "index.ts", "web.ts"],
+  "include": ["src/**/*", "jest.config.ts", ".eslintrc.cjs", "index.ts"],
```
Comment (petersalas) [inline packages/fixie/tsconfig.json:17]: I'm pretty sure you're breaking into jail w.r.t. ESM/CJS/TS because Next.js apps default to CJS -- at the very least you should do this as part of a different change

File: packages/fixie/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export * from './client.js';
+export * from './fixie-embed.js';
```
Comment (petersalas) [inline packages/fixie/src/index.ts:2]: I'm not sure about combining `fixie` and `fixie/web` because it will require React everywhere

**Comment by mdwelsh** [comment]
Okay! Thanks for the comments. Reverted everything but the specific change to the user stuff, but also renamed sidekick-types.ts to types.ts since I put the User type in there too.

File: packages/fixie/src/agent.ts [hunk 4]
```diff
@@ -500,7 +500,7 @@ export class FixieAgent {
     environmentVariables: Record<string, string> = {}
   ): Promise<AgentRevision> {
     const config = await FixieAgent.LoadConfig(agentPath);
-    const agentId = `${(await client.userInfo()).username}/${config.handle}`;
+    const agentId = config.handle;
```
Comment (petersalas) [inline packages/fixie/src/agent.ts:503]: This is a misleading variable name -- can you remove the variable and just use `config.handle` everywhere?

File: packages/fixie/src/agent.ts [hunk 5]
```diff
@@ -549,7 +549,7 @@ export class FixieAgent {
     debug?: boolean;
   }) {
     const config = await FixieAgent.LoadConfig(agentPath);
-    const agentId = `${(await client.userInfo()).username}/${config.handle}`;
+    const agentId = config.handle;
```
Comment (petersalas) [inline packages/fixie/src/agent.ts:552]: Ditto

File: packages/fixie/src/main.ts [hunk 6]
```diff
@@ -161,6 +163,19 @@ program
     })
   );
 
+user
+  .command('update')
+  .description('Update information on the current user')
+  .option('--email <string>', 'The new email address for this user')
+  .option('--fullName <string>', 'The new description for this corpus')
```
Comment (petersalas) [inline packages/fixie/src/main.ts:170]: bad copy/paste in the arg help

**Review by petersalas — APPROVED** [review]

**Comment by mdwelsh** [comment]
### Merge activity

* **Nov 9, 9:47 PM**: @mdwelsh merged this pull request with [Graphite](https://app.graphite.dev/github/pr/fixie-ai/ai-jsx/487).
